mutation/mutation_fragment.hh

```diff
@@ -32,6 +32,7 @@
 class clustering_row {
     clustering_key_prefix _ck;
     deletable_row _row;
+    int32_t _merge_count{ 1 };
```

Comment:
Why is the statistic in the row itself?

Reply:
That is simple enough, the tricky part is propagating up this statistics though all the reader layers, to the compaction one. The `clustering_row` is a natural place to store this IMO.

Reply:
No, clustering_row has nothing to do with counting merges during compaction.

Propagating the statistics structure through the layers is annoying, but doesn't inject transient data where it doesn't belong. Don't be lazy.